index.py

```diff
@@ -214,56 +267,64 @@ def on_message(client, userdata, message):
-        return
+        return    
 
     if plate_number is None:
         _LOGGER.info(f'No plate number found for event {frigate_event}')
-        return
+        is_valid_plate = False
 
     # check score
     min_score = config['frigate'].get('min_score')
-    if min_score and score < min_score:
-        _LOGGER.info(f"Score is below minimum: {score}")
-        return
-
-    
-
-    # get db connection
-    conn = sqlite3.connect(DB_PATH)
-    cursor = conn.cursor()
-
-    # Insert a new record of plate number
-    _LOGGER.info(f"Storing plate number in database: {plate_number} with score: {score}")
+    if min_score and score:
+        if score < min_score:
+            _LOGGER.info(f"Score is below minimum: {score}")
+            is_valid_plate = False
 
-    start_time = datetime.fromtimestamp(after_data['start_time'])
-    formatted_start_time = start_time.strftime("%Y-%m-%d %H:%M:%S")
     
-    cursor.execute("""
-        INSERT INTO plates (detection_time, score, plate_number, frigate_event, camera_name) VALUES (?, ?, ?, ?, ?)
-    """, (formatted_start_time, score, plate_number, frigate_event, after_data['camera']))
-    conn.commit()
-    conn.close()
-
-    # set the sublabel
-    set_sublabel(frigate_url, frigate_event, plate_number, score)
-
-    # send mqtt message
-    if config['frigate'].get('return_topic'):
-        send_mqtt_message({
-            'plate_number': plate_number,
-            'score': score,
-            'frigate_event': frigate_event,
-            'camera_name': after_data['camera'],
-            'start_time': formatted_start_time
-        })
-
+    if is_valid_plate:
+        # get db connection
+        conn = sqlite3.connect(DB_PATH)
+        cursor = conn.cursor()
+
+        # Insert a new record of plate number
+        _LOGGER.info(f"Storing plate number in database: {plate_number} with score: {score}")
+
+        start_time = datetime.fromtimestamp(after_data['start_time'])
+        formatted_start_time = start_time.strftime("%Y-%m-%d %H:%M:%S")
+        
+        cursor.execute("""
+            INSERT INTO plates (detection_time, score, plate_number, frigate_event, camera_name) VALUES (?, ?, ?, ?, ?)
+        """, (formatted_start_time, score, plate_number, frigate_event, after_data['camera']))
+        conn.commit()
+        conn.close()
+
+        # set the sublabel
+        set_sublabel(frigate_url, frigate_event, plate_number, score)
+
+        # send mqtt message
+        if config['frigate'].get('return_topic'):
+            send_mqtt_message({
+                'plate_number': plate_number,
+                'score': score,
+                'frigate_event': frigate_event,
+                'camera_name': after_data['camera'],
+                'start_time': formatted_start_time
+            })
+            
+    # save image
+    if config['frigate']['save_snapshots']:
+        if (plate_number is not None and min_score and score > min_score) or config['frigate']['always_save_latest_file']:
```

Comment:
use get here too for `always_save_latest_file`

Reply:
i see the code above that instead of returning when invalid license plate for min score, you set a flag but then check for those here anyways. Since we check for these anyways we can get rid of the `is_valid_plate` flag completely and return like normal then this check simplifies to

```
if config['frigate'].get('always_save_latest_file', False):
```

Reply:
The reason I did it this way is because if there is a missing plate or min score, we don't want the routine to return too early. If `always_save_latest_file: True`, we need the routine to run all the way to line 329 to save the 'empty' snapshot.  If we set e.g. line 208 back to `return`, `on_message` routine would return before ever getting to `save_image`. Or am I missing something?